slides/k8s/helm-dependencies.md

```diff
@@ -1,135 +1,275 @@
-# Creating even better charts
+# Charts using other charts
 
-- We will going to use dependencies to write less code
+- Helm charts can have *dependencies* on other charts
 
-- reuse a community chart for redis
+- These dependencies will help us to share or reuse* components
 
-- We will see how it improve the developer life as well as the one of the user
+  (so that we write and maintain less manifests, less templates, less code!)
 
-- We will see that sometimes dependencies could also remove a lot of code
+- As an example, we will use a community chart for Redis
+
+- This will help people who write charts, and people who use them
+
+- ... And potentially remove a lot of code! ✌️
 
 ---
 
-## Assertions about dockercoins
+## Redis in DockerCoins
+
+- In the DockerCoins demo app, we have 5 components:
+
+  - 2 internal webservices
+  - 1 worker
+  - 1 public web UI
+  - 1 Redis data store
 
-- Every image is custom except the one of redis
+- Every component is running some custom code, except Redis
 
-- It's is quite logic that deployment are custom except of the one of redis
+- Every component is using a custom image, except Redis
 
-- So if redis is not custom can we use templates of someone else ?
+  (which is using the official `redis` image)
 
-- Yes, through dependencies
+- Could we use a standard chart for Redis?
+
+- Yes! Dependencies to the rescue!
 
 ---
 
-## Add redis dependency
+## Adding our dependency
+
+- First, we will add the dependency to the `Chart.yaml` file
+
+- Then, we will ask Helm to download that dependency
+
+- We will also *lock* the dependency
+
+  (lock it to a specific version, to ensure reproducibility)
+
+---
+
+## Declaring the dependency
+
+- First, let's edit `Chart.yaml`
 
-- In `Chart.yaml` fill the `dependencies` section:
+.exercise[
+
+- In `Chart.yaml`, fill the `dependencies` section:
   ```yaml
     dependencies:
       - name: redis
         version: 11.0.5
         repository: https://charts.bitnami.com/bitnami
         condition: redis.enabled
   ```
-- `condition` will trigger at install or upgrade to template or not the dependency
 
-- Now, we can tell helm to fetch the dependency by running `helm dependency update`
+]
+
+Where do that `repository` and `version` come from?
+
+We're assuming here that we did our reserach,
+or that our resident Helm expert advised us to
+use Bitnami's Redis chart.
+
+---
+
+## Conditions
+
+- The `condition` field gives us a way to enable/disable the dependency:
+  ```yaml
+  conditions: redis.enabled
+  ```
+
+- Here, we can disable Redis with the Helm flag `--set redis.enabled=false`
+
+  (or set that value in a `values.yaml` file)
 
-  (abbv `helm dep up`)
+- Of course, this is mostly useful for *optional* dependencies
```

Comment:
In real life, I personally never used conditions for optional dependencies but only to swap the dependency chart by something else (eg. a DB, by an operator of that DB, or a cloud instance)

Reply:
I'm trying to put together a list a good practice/feature for developping helm chart. Add this will definitively in